### PR TITLE
Add a formula for Jack2

### PR DESCRIPTION
**Pull request by ColinKinloch**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

`jack` c++ rewrite maintained in parallel with `jack`. Has support for CoreMidi on MacOS.
Mentioned in a comment in the `jack` formula added in 2013.

https://github.com/Homebrew/homebrew-core/blob/fe7334d3c60777926ec25d42e40546601ea3c662/Formula/jack.rb#L2-L6

I don't know if this still applies.

Is there anything missing? Does this formula (`jack2`) belongs in `homebrew-core`?

### CODE REVIEW

File: Formula/jack2.rb [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        <true/>
+        <key>KeepAlive</key>
+        <true/>
+      </dict>
+      </plist>
+    EOS
+  end
+
+  test do
+    assert_match /#{version}/, shell_output("#{bin}/jackd --version", 255)
```
Comment (fxcoudert) [inline Formula/jack2.rb]: We need a test that exercises the some of the functionality of the app.

**Review by fxcoudert — CHANGES_REQUESTED** [review]

**Comment by ColinKinloch** [comment]
So, what is the next step?

**Comment by fxcoudert** [comment]
@ColinKinloch we reopen it :)

My question about this is: if this is a really a reimplementation of jack and maintained in parallel, why is it conflicting? Because in the current status, given it's a library, it means other formulas will want to depend on it, and it will increase the number of conflicts between jack and jack2-linked formulas, since you can't have both on your system

**Comment by ColinKinloch** [comment]
jack and jack2 are interchangable from the point of view of the linked clients.

The benefit of jack2 on a MacOS system is support for CoreMIDI as a backend.

[FAQ: differences between JACK1 and JACK2](https://github.com/jackaudio/jackaudio.github.com/wiki/Q_difference_jack1_jack2)

If you compare repos for [jack1](https://github.com/jackaudio/jack1) and [jack2](https://github.com/jackaudio/jack2), jack2 seems to be under more active development, though I'm not sure what this indicates about the relative health of the projects.

**Comment by ColinKinloch** [comment]
Thinking about it, what I should be suggesting is having the `jack` formula install jack2 rather than jack1.

**Pull request closed without merging**